specification/hugr.md

```diff
@@ -665,68 +674,103 @@ See [Type System](#type-system) for more on Resources.
 # may need some top level data, e.g. namespace?
 
 # Import other header files to use their custom types
+  # TODO: allow qualified, and maybe locally-scoped
```

Comment:
@croyzor I'm a bit unclear how this works. We've imported Quantum, and we say `Q` is defined there - but, as an alias? Or an `Opaque`? Those are, AFAI am aware, the only two ways for an extension to define new types. Maybe alias works, or maybe `Q` is not a good example.

Elsewhere (below) I've been writing `Opaque(complex_matrix,...)` but maybe that should just be `complex_matrix`?? and that should be imported from another resource - in which case, lets do a qualified import for that....